### PR TITLE
CLOUD-349, CLOUD-14: Document setup of Dashboard GitHub Enterprise Integration

### PR DESCRIPTION
**Pull request by debrisapron**
This is a draft of the instructions for setting up the new Dashboard GitHub Enterprise integration. There are some aspects that still need to be resolved:

- [x] The screenshots need to be added. I have already created all the images needed, but I noticed that the existing screenshots on this page have a consistent style of annotation which is out of my limited design abilities to exactly reproduce. Is it ok for my screenshots to be inconsistent with the existing ones, do I need to hand them to a designer, or can someone tell me how to get this exact style?
- [x] Once the screenshots are in place they need anchors for the Dashboard page to link to.
- [x] Is it too much to have both sets of instructions on a single page? Should we spin out GitHub Enterprise into its own page in the Dashboard section?
- [x] Without seeing the final GitHub Enterprise configuration page on the Dashboard, I can't confirm 100% that my instructions for filling it in are correct.

### CODE REVIEW

File: content/guides/dashboard/github-integration.md [hunk 1]
```diff
@@ -88,6 +107,78 @@ you choose **All repositories**.
 
 3. Click the **Install** button to complete the installation.
 
+### Cypress GitHub Enterprise app installation process
+
+For GitHub Enterprise the installation is a little more involved, you need to
+create a new GitHub App, copy various settings and credentials from the new app,
+paste them into the Dashboard and go through the GitHub app activation process.
+
```
Comment (cy-veronica) [inline content/guides/dashboard/github-integration.md:115]: Run on sentence. Suggest...
To install GitHub Enterprise Integration you need to; create a new GitHub App, copy the required settings and credentials from the new app to the Cypress configuration form, and complete the GitHub app activation process.

File: content/guides/dashboard/github-integration.md [hunk 2]
```diff
@@ -88,6 +107,78 @@ you choose **All repositories**.
...
+     numbers, **no spaces**.
+   - Enter the Cypress Dashboard **Homepage URL**, https://dashboard.cypress.io
+   - Enter the **Callback URL** and **Setup URL**,
+     https://dashboard.cypress.io/apps/github/callback
+
+3. Fill in the **Webhook** form.
+
+   - Enter the **Webhook URL**, https://dashboard.cypress.io/webhooks/github-app
+   - Copy and paste your Cypress organization's **Webhook secret** from the
+     Dashboard into the GitHub Webhook form.
```
Comment (cy-veronica) [inline content/guides/dashboard/github-integration.md:134]: You can copy this directly from the Dashboard GitHub Enterprise configuration page.

File: content/guides/dashboard/github-integration.md [hunk 3]
```diff
@@ -88,6 +107,78 @@ you choose **All repositories**.
...
+
+   - Enter a **GitHub App name**. Name may contain only dashes, letters and
+     numbers, **no spaces**.
+   - Enter the Cypress Dashboard **Homepage URL**, https://dashboard.cypress.io
+   - Enter the **Callback URL** and **Setup URL**,
+     https://dashboard.cypress.io/apps/github/callback
+
+3. Fill in the **Webhook** form.
+
+   - Enter the **Webhook URL**, https://dashboard.cypress.io/webhooks/github-app
```
Comment (cy-veronica) [inline content/guides/dashboard/github-integration.md:132]: You can copy this directly from the Dashboard GitHub Enterprise configuration page.

**Review by cy-veronica — CHANGES_REQUESTED** [review]
Please include your screenshots. We have enough consistency issues that I am not worried about them being different that the instructions for GitHub cloud.

File: content/guides/dashboard/github-integration.md [hunk 4]
```diff
@@ -88,6 +107,118 @@ you choose **All repositories**.
...
+   - Copy and paste the root URL of your GitHub Enterprise site into the
+     **GitHub Enterprise Server Url** field.
+
+<DocsImage src="/img/dashboard/github-integration/ghe/ghe-05.png" alt="Collect new app info" ></DocsImage>
+
+<a id="cypress-github-enterprise-app-installation-process-6"></a>
+
+6. Click the **Generate a new client secret** button, then copy and paste the
+   generated secret into the **GitHub Enterprise client secret** field on the
+   Dashboard.
```
Comment (cy-veronica) [inline content/guides/dashboard/github-integration.md:179]:
```suggestion
```

File: content/guides/dashboard/github-integration.md [hunk 5]
```diff
@@ -88,6 +107,118 @@ you choose **All repositories**.
...
+
+<DocsImage src="/img/dashboard/github-integration/ghe/ghe-06.png" alt="Generate client secret" ></DocsImage>
+
+<a id="cypress-github-enterprise-app-installation-process-7"></a>
+
+7. Scroll to the bottom of the app settings page and click the **Generate a
+   private key** button. A file will be downloaded to your browser's default
+   downloads location. Open the file with a plain text editor, select all text
+   and copy and paste into the **GitHub Enterprise private key** field on the
+   Dashboard.
```
Comment (cy-veronica) [inline content/guides/dashboard/github-integration.md:189]:
```suggestion
```

File: content/guides/dashboard/github-integration.md [hunk 6]
```diff
@@ -88,6 +107,118 @@ you choose **All repositories**.
...
+
+<DocsImage src="/img/dashboard/github-integration/ghe/ghe-07.png" alt="Generate private key" ></DocsImage>
+
+<a id="cypress-github-enterprise-app-installation-process-8"></a>
+
+8. In the Dashboard, click the **Next Step** button and you will be taken to the
```
Comment (cy-veronica) [inline content/guides/dashboard/github-integration.md:195]:
```suggestion
7. In the Dashboard, click the **Next Step** button and you will be taken to the
```

File: content/guides/dashboard/github-integration.md [hunk 7]
```diff
@@ -88,6 +107,118 @@ you choose **All repositories**.
...
+   - Copy and paste the App name into the **GitHub Enterprise app name** field.
+   - Copy and paste the App ID into the **GitHub Enterprise app ID** field.
+   - Copy and paste the Client ID into the **GitHub Enterprise client ID**
+     field.
+   - Copy and paste the root URL of your GitHub Enterprise site into the
+     **GitHub Enterprise Server Url** field.
+
+<DocsImage src="/img/dashboard/github-integration/ghe/ghe-05.png" alt="Collect new app info" ></DocsImage>
+
+<a id="cypress-github-enterprise-app-installation-process-6"></a>
```
Comment (cy-veronica) [inline content/guides/dashboard/github-integration.md:175]: This screenshot should include the "GitHub App name" field with a highlight.

File: content/guides/dashboard/github-integration.md [hunk 8]
```diff
@@ -88,6 +107,118 @@ you choose **All repositories**.
...
+
+8. In the Dashboard, click the **Next Step** button and you will be taken to the
+   GitHub Enterprise app authorization page. Click the **Authorize \[your app
+   name\]** button.
+
+<DocsImage src="/img/dashboard/github-integration/ghe/ghe-08.png" alt="Authorize GitHub App" ></DocsImage>
+
+<a id="cypress-github-enterprise-app-installation-process-9"></a>
+
+9. On your newly-authorized GitHub App, click the **Install** button.
```
Comment (cy-veronica) [inline content/guides/dashboard/github-integration.md:203]:
```suggestion
8. On your newly-authorized GitHub App, click the **Install** button.
```

File: content/guides/dashboard/github-integration.md [hunk 9]
```diff
@@ -88,6 +107,118 @@ you choose **All repositories**.
...
+
+<DocsImage src="/img/dashboard/github-integration/ghe/ghe-09.png" alt="Install GitHub App" ></DocsImage>
+
+<a id="cypress-github-enterprise-app-installation-process-10"></a>
+
+10. Nearly there! On the GitHub App installation page, choose whether you want
```
Comment (cy-veronica) [inline content/guides/dashboard/github-integration.md:209]:
```suggestion
9. Nearly there! On the GitHub App installation page, choose whether you want
```

File: content/guides/dashboard/github-integration.md [hunk 10]
```diff
@@ -88,6 +107,118 @@ you choose **All repositories**.
...
+    **Install** button.
+
+<DocsImage src="/img/dashboard/github-integration/ghe/ghe-10.png" alt="Confirm installation of GitHub App" ></DocsImage>
+
+<a id="cypress-github-enterprise-app-installation-process-"></a>
+
+11. Finally you will be returned to the Dashboard. Congratulations, you have
+    installed the Cypress Dashboard GitHub Enterprise integration! Treat
+    yourself to the refreshing beverage of your choice, then continue to the
+    next section.
```
Comment (cy-veronica) [inline content/guides/dashboard/github-integration.md:220]:
```suggestion
10. Finally, you will be returned to the Dashboard. Congratulations, you have
    installed the Cypress Dashboard GitHub Enterprise integration! You are now
    ready configure a GitHub Integration for a project.
```

File: content/guides/dashboard/github-integration.md [hunk 11]
```diff
@@ -96,12 +227,12 @@ can now enable GitHub Integration for _any_ Cypress project.
 1. Go to the project's settings page.
    <DocsImage src="/img/dashboard/visit-project-settings.png" alt="Visit project settings" ></DocsImage>
 
-2. Scroll down to the GitHub Integration section. <Alert type="info">
+2. Scroll down to the GitHub Integration section.
```
Comment (cy-veronica) [inline content/guides/dashboard/github-integration.md:230]:
```suggestion
2. Scroll down to the GitHub Integration or GitHub Enterprise Integration section.
```

**Review by cy-veronica — DISMISSED** [review]
Please verify that my use of Suggest did not mess up the formatting (indents and numbering specifically).

File: content/guides/dashboard/github-integration.md [hunk 12]
```diff
@@ -74,20 +91,102 @@ to GitHub.com to complete the installation:
...
+   Your new GitHub App is now created, and you'll be taken to the settings page.
+
+6. Returning to the Cypress Dashboard GitHub Enterprise Integration
+   configuration screen, populate the following values from your new GitHub App
+   - Copy the root URL of your GitHub Enterprise site to the Cypress **GitHub
+     Enterprise Server Url** field.
+   - Copy the **App ID** to the Cypress **GitHub Enterprise App ID** field.
+   - Copy the **Client ID** to the Cypress **GitHub Enterprise Client ID**
+     field.
+   - Copy the **GitHub App name** to the Cypress **GitHub Enterprise app name**
```
Comment (cy-veronica) [inline content/guides/dashboard/github-integration.md:155]:
```suggestion
   - Copy the **GitHub App name** to the Cypress **GitHub Enterprise App name**
```

File: content/guides/dashboard/github-integration.md [hunk 13]
```diff
@@ -74,20 +91,102 @@ to GitHub.com to complete the installation:
...
+
+<DocsImage src="/img/dashboard/github-integration/ghe/ghe-04.png" alt="Configure app permissions" id="cypress-github-enterprise-app-installation-process-4"></DocsImage>
+
+5. Skip to the bottom of the form and click the **Create GitHub App** button.
+   Your new GitHub App is now created, and you'll be taken to the settings page.
+
+6. Returning to the Cypress Dashboard GitHub Enterprise Integration
+   configuration screen, populate the following values from your new GitHub App
+   - Copy the root URL of your GitHub Enterprise site to the Cypress **GitHub
+     Enterprise Server Url** field.
```
Comment (cy-veronica) [inline content/guides/dashboard/github-integration.md:151]:
```suggestion
     Enterprise Server URL** field.
```

File: content/guides/dashboard/github-integration.md (hunk 12 above)
Comment (cy-veronica) [inline content/guides/dashboard/github-integration.md:155]: Still need to include a screenshot of the GitHub App name field from GitHub

**Review by cy-veronica — CHANGES_REQUESTED** [review]

File: content/guides/dashboard/github-integration.md (hunk 12 above)
Comment (debrisapron) [reply content/guides/dashboard/github-integration.md:155]: It's already there https://cypress-documentation-n6c1j1liv-cypress-io.vercel.app/guides/dashboard/github-integration#cypress-github-enterprise-app-installation-process-5

File: content/guides/dashboard/github-integration.md (hunk 13 above)
Comment (debrisapron) [reply content/guides/dashboard/github-integration.md:151]: That's how the field is written in the Dashboard, are you sure you want to change it? We would need to change the dashboard to match.

File: content/guides/dashboard/github-integration.md (hunk 12 above)
Comment (debrisapron) [reply content/guides/dashboard/github-integration.md:155]: That's how the field is written in the Dashboard, are you sure you want to change it? We would need to change the dashboard to match.

File: content/guides/dashboard/github-integration.md (hunk 13 above)
Comment (cy-veronica) [reply content/guides/dashboard/github-integration.md:151]: Yes, it is being corrected as part of a separate Jira ticket to correctly reflect GitHub.

File: content/guides/dashboard/github-integration.md (hunk 12 above)
Comment (cy-veronica) [reply content/guides/dashboard/github-integration.md:155]: Yes, it is being corrected as part of a separate Jira ticket to correctly reflect GitHub.

Comment (cy-veronica) [reply content/guides/dashboard/github-integration.md:155]: See it now. Was expecting the field, not the header, but that works. Will resolve.

**Review by cy-veronica — APPROVED** [review]
Looks good!

File: content/guides/dashboard/github-integration.md [hunk 14]
```diff
@@ -74,20 +91,102 @@ to GitHub.com to complete the installation:
...
+   - **Pull requests**: read & write
+   - **Commit statuses**: read & write
+
+<DocsImage src="/img/dashboard/github-integration/ghe/ghe-04.png" alt="Configure app permissions" id="cypress-github-enterprise-app-installation-process-4"></DocsImage>
+
+5. Skip to the bottom of the form and click the **Create GitHub App** button.
+   Your new GitHub App is now created, and you'll be taken to the settings page.
+
+6. Returning to the Cypress Dashboard GitHub Enterprise Integration
+   configuration screen, populate the following values from your new GitHub App
```
Comment (jaffrepaul) [inline content/guides/dashboard/github-integration.md:149]:
```suggestion
   configuration screen, populate the following values from your new GitHub App.
```

**Review by jaffrepaul — CHANGES_REQUESTED** [review]
I added one tiny punctuation suggestion. Otherwise LGTM!

File: content/guides/dashboard/github-integration.md (hunk 14 above)
Comment (debrisapron) [reply content/guides/dashboard/github-integration.md:149]: Good catch!

Comment (debrisapron) [reply content/guides/dashboard/github-integration.md]: Resolved by https://github.com/cypress-io/cypress-documentation/pull/4336/commits/19bdd151fb95bca0b6b112d75de49fccf9f689c3#diff-031fb3f5fe6eb378cdd54e4a238398cf73ff1afa6543ae24741a3d2be8e85cd0R149

**Review by jaffrepaul — APPROVED** [review]
LGTM 🎉

**Pull request merged**